src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,204 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function gpkgread(fname; layer=1)
+  db = SQLite.DB(fname)
+  assertgpkg(db)
+  geoms = gpkggeoms(db; layer)
+  table = gpkgvalues(db; layer)
+  DBInterface.close!(db)
+  if eltype(table) <: Nothing
+    return georef(nothing, geoms)
+  else
+    georef(table, geoms)
+  end
+end
+
+function assertgpkg(db)
+
+  # Requirement 6: PRAGMA integrity_check returns a single row with the value 'ok'
+  # Requirement 7: PRAGMA foreign_key_check (w/ no parameter value) returns an empty result set
+  if first(DBInterface.execute(db, "PRAGMA integrity_check;")).integrity_check != "ok" ||
+     !(isempty(DBInterface.execute(db, "PRAGMA foreign_key_check;")))
+    throw(ErrorException("database integrity at risk or foreign key violation(s)"))
+  end
+
+  # Requirement 10: must include a gpkg_spatial_ref_sys table
+  # Requirement 13: must include a gpkg_contents table
+  tbcount = first(DBInterface.execute(
+    db,
+    """
+  SELECT COUNT(*) AS n FROM sqlite_master WHERE 
+  name IN ('gpkg_spatial_ref_sys', 'gpkg_contents') AND 
+  type IN ('table', 'view');
+    """
+  ))
+  if tbcount.n != 2
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+end
+
+function gpkgvalues(db, ; layer=1)
```

Comment:
This function should return a single Tables.jl table or nothing as explained in previous reviews. That way we can simply forward the result to `georef` without additional checks.